docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6324,380 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
 `p4.v1.TableEntry` message based on the value of architecture-specific table
 properties, but we may include on in future versions of the API.
 
+# GenericTable { #sec-generic-table}
+
+## GenericTable p4info { #sec-p4info-generic-table}
+
+`GenericTable` messages are used to map non-PSA P4-externs or non-P4 target-
+specific "fixed" features to their implementation in a generic way. The same can
+be achieved via `Extern`, but GenericTable provides a structured way in which
+every feature is represented as a set of match-fields and data-fields. The data
+consists of unions which are similar to actions. One use of GenericTable in a
+server backend is with TDI [Table Driven
+Interface](https://github.com/p4lang/tdi), but targets can use Generic table to
+map to their own specific feature implementations as well.  It defines the
+following fields
+
+* `generic_table_type_id`, an 8-bit unsigned integer which uniquely identifies
+  the generic_table_type in the context of the architecture.
+  This value is in the range of `[0x00, 0xff]`.
+  The ID in the preamble is created as `0x18 + generic_table_type_id + 16 bits
+  of generated ID by compiler`. The `generic_table_type` depicts a new table
+  "type" which would have similar properties. The organization of the table
+  type space is left up to the target. Some optional guidelines
+  * For P4 externs, any new extern to have a new table type.
+  * For non-P4 externs, options are
```

Comment:
If by "non-P4 externs" you mean "target-specific fixed function blocks" I would recommend using the latter phrase, to be consistent with earlier usage.

Also I personally find the phrase "non-P4 extern" to be just about a contradiction in terms.  `extern` is a P4 language construct.

Reply:
The list of detailed recommendations on how an implementer might manage the generic_table_type id space seems to interrupt the flow here.  Consider putting that in a separate small section focused on that purpose, perhaps?

Reply:
Note: If you create a separate small section focused on managing generic_table_type id, it would also be reasonable to put one or two examples of the different approaches you describe there, to make them easier to understand.

Reply:
Added a new section with some examples